oryx-common/src/lib.rs

```diff
@@ -2,27 +2,46 @@
 
 use core::mem;
 
-use network_types::{arp::ArpHdr, icmp::IcmpHdr, ip::IpHdr, tcp::TcpHdr, udp::UdpHdr};
+use network_types::{arp::ArpHdr, eth::EthHdr, icmp::IcmpHdr, ip::IpHdr, tcp::TcpHdr, udp::UdpHdr};
 
 pub mod protocols;
 
 pub const MAX_FIREWALL_RULES: u32 = 32;
 pub const MAX_RULES_PORT: usize = 32;
 
+#[repr(C)]
+#[derive(Clone)]
+pub struct RawFrame {
+    pub header: EthHdr,
```

Comment:
maybe we should define the header here with the fields we are interested in to be more explicit, what do you think ?

Reply:
Eh it would be easy enough to do this, I dont have a strong opinion either way. My thinking in using the already defined type was mostly for convenience since it is already defined and is returned from the `TcContext` so app layer dosnt need to do extra work to convert it (not a big deal to implement a `From` impl though). But also it includes additional information that could be interesting to render later, mainly in the eth type. But this assumes `network_types` expands the eth type enum to cover other interesting packet types like Wake on LAN, MPLS, or VLAN etc., but that would of course need changes in `aya`